xonsh/environ.py

```diff
@@ -120,6 +121,7 @@ def to_debug(x):
     'XONSH_ENCODING_ERRORS': (is_string, ensure_string, ensure_string),
     'XONSH_HISTORY_SIZE': (is_history_tuple, to_history_tuple, history_tuple_to_str),
     'XONSH_LOGIN': (is_bool, to_bool, bool_to_str),
+    'XONSH_SHOW_TRACEBACK': (is_tb_opt, to_tb_tuple, tb_tuple_to_str),
```

Comment:
@adqm  Well, it turns out there are a few corner cases.

If a user has specified:
```
$XONSH_SHOW_TRACEBACK = True
```

After detyping, it becomes `'True'`, which is also a perfectly valid filename, and it is not clear whether retyping `'True'` should produce the boolean value or the filename itself.

To avoid such ambiguities, I think it is necessary to either use a structured representation like a tuple or introduce another env. variable that refers exclusively to the log output file.